parlai/agents/hugging_face/gpt2.py

```diff
@@ -35,12 +34,21 @@ def __init__(self, opt, dict):
         super().__init__()
         self.transformer = self._init_from_pretrained(opt)
         # add special tokens
-        self.start_idx = dict.start_idx
-        self.null_idx = dict.null_idx
-        self.add_start_token = False
         if opt["add_special_tokens"]:
+            size_before = self.transformer.wte.weight.size(0)
             self.transformer.resize_token_embeddings(len(dict.tokenizer))
-            self.add_start_token = opt["add_start_token"]
+            with torch.no_grad():
```

Comment:
Wondering as to why we need this ?

Reply:
The torch nograd or the initialization changes?